mm2src/coins/qrc20.rs

```diff
@@ -891,6 +965,47 @@ impl MmCoin for Qrc20Coin {
         // QRC20 cannot have unspendable balance
         Box::new(futures01::future::ok(0.into()))
     }
+
+    fn swap_contract_address(&self) -> Option<BytesJson> {
+        Some(BytesJson::from(self.swap_contract_address.0.as_ref()))
+    }
+}
+
+async fn qrc20_rpc_contract_call(
+    rpc_client: &UtxoRpcClientEnum,
+    func: RpcContractCallType,
+    contract_addr: &H160,
+    tokens: &[Token],
+) -> Result<Vec<Token>, String> {
+    let function = func.as_function();
+    let params = try_s!(function.encode_input(tokens));
+    let contract_addr = contract_addr_into_rpc_format(contract_addr);
+
+    let result: ContractCallResult = match rpc_client {
+        UtxoRpcClientEnum::Native(native) => try_s!(native.call_contract(&contract_addr, params.into()).compat().await),
```

Comment:
I think that this and other match statements can be extracted to the `Qrc20RpcOps` trait.

Reply:
Done, thanks, it looks better than static functions like `qrc20_rpc_contract_call` and `request_token_decimals`.